### DISCUSSION TITLE
Ender 3 S1 UBL & 6x6 Mesh

### DISCUSSION THREAD

**Discussion by I-wish-2** [mriscoc/Ender3V2S1#779] [Q&A]
Greetings,
I too was having big problems with my E3 S1 regarding UBL and a usable 6x6 mesh.  I have read most all of the discussions regarding the S1 and mesh issues here and I thought I might share what has worked for me to resolve the mesh probing problems.

A little back story,  My S1 is an eBay purchase from a print farm that needed to be rebuilt.  It had the original Creality motherboard, the F301, which was toast.  Because the F301 (512k) board is no longer available and the F401 (256k) is all I could find which limits the features of Miguel's firmware.  I installed a Creality 4.2.7 board and with some extensive rewiring of the sprite head I powered up the S1 and flashed it with Miguel's current firmware the display was replaced with an E3V2 Dwin flashed unit.  After all the initial setup and calibrations I was on to the Tramming, Z Offset and bed mesh.
The tramming went great will all points tweaked to +/- 0.01mm, the Z offset was set to -1.02mm with a .15mm feeler Gauge but the Mesh was just a nightmare with a range of about 4.00mm.

This E3 S1 is the 7th ender i have and I have never seen such a mess with trying to produce a usable mesh.  After much research and the views expressed on this forum, Thank You All, I found that the main focus was directed at the mechanical set up of the X,Y & Z axis and all of their components.  In addition there were notes about X & E stepper inducing noise into the CR Touch as well as the X & Y probe offset values.

I set about resolving any and all mechanical issues from wheels to cams and bolts bearings and screws. I adjusted and verified  the gantry vertical alignment on the right and left side to with in 0.25mm measured to the frame.  Then tramming for the bed, Z Offset once that was completed I tested the mesh again and nope it was still a hopeless mess.

I then re-read the comments about the Probe X & Y offsets and that they might causing issues as they were over 45mm each.  I looked on Printables.com for this probe mount, https://www.printables.com/model/257460-ender-3-s1-zero-offset-cr-touch-mount, and then under the Remixes for the one that maker "zosman" made, (https://www.printables.com/model/405829-ender3s1crtouchmountstronger)
I printed and installed this Mount for the CR Touch ( remove the old fan and CR Touch bracket. I then edited the probe offsets for X = -47.5 and Y = 0.00.

I re-ran the Tramming, Z-Offset and then the 6 x 6 Mesh and the results were stunning.  The range was now -0.04 to 0.12mm over 36 points.
I then tested the mesh by using the Edit Mesh feature and went to each of the 36 points and tested the nozzle to bed spacing with the 0.15mm gauge and each one of the X & Y points was right on the money.
At the present I'm printing all over the Kingroon PEX flex bed the various upgrade and enhancement parts I found on Printables and they are all have great first layer adhesion.

I want to thank everyone here that helped me resolve my UBL Mesh issues and can only suggest that you give the CR Touch mount a try if your still looking for an answer and I hope it will help you get on to printing.  Thank for reading and Sorry for being so long winded. 
Regards
Doug

**Comment by I-wish-2** [comment]
From Printables.com I found a refined and robust CR Touch bracket that is non invasive to the Sprite.  I'm testing it now as if reduces the Probe X offset by about 5mm.  It attaches with out removing Sprite parts.
https://www.printables.com/model/259877-ender-s1-sprite-extruder-cr-touch-mount
In addition I removed the lower half of the Hot End fan shroud for better air flow with this version.
For higher temp filaments the use of an infill of 80% might be better.
Doug

**Reply by I-wish-2** [reply]
Some follow up on the final revision and results.
Regards
Doug
![Sprite 1](https://user-images.githubusercontent.com/103605403/223495617-9cfa7e78-2d04-4781-9141-a00cc9b9a330.jpg)
![Sprite 2](https://user-images.githubusercontent.com/103605403/223495671-5a2fda91-97c6-4976-9619-8547b9a60302.jpg)
![Slot 0](https://user-images.githubusercontent.com/103605403/223495699-fa17df21-adef-4663-8d1f-f327d9aa0879.jpg)

**Comment by ghost** [comment]
so, basically this is about that default CT touch is not working fine, when default offset for Ender 3 S1? And making Y = 0 X about 40mm with this printable, was ok?

Is there any possibility default handler wasn't in tact with the geometrical shape of it, so the CR touch was not measuring right?

Have you measured the printable thing you did so it is in some tolerance anyone could later say is is ok?

Btw. how do you measure the offsets from CR Touch and nozzle? I find it very hard not to make triangle error.

**Reply by I-wish-2** [reply]
1. yes, I would say that the factory Sprite probe layout is not the optimized.  With the change to the mount and y = 0 and x = 47.5 the mesh build was good as was the printing.
> Is there any possibility default handler wasn't in tact with the geometrical shape of it, so the CR touch was not measuring right?
I'm not sure I understand your comment.
3.  I'm installing an new SKR Mini E3 V3.0 today and will build an 8 x 8 mesh, HS mode is disabled, The I will run the attached STL file to verify the complete bed area.  This file is scaled to 190 x 190 to fit the S1 bed.  Then some cubes to verify all the X,Y & Z axis's are in calibration.
4. Measuring the offsets with the Sprite is a little easer as it can be removed and placed in a square jig.  Measurements with a micrometer can then be made quite accurately.

I will post the results of the prints when the upgrade is completed.
Doug
